File: presentation/src/main/java/com/example/presentation/products/adapter/ProductsAdapter.java
package com.example.presentation.products.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.presentation.R;
import com.example.presentation.model.ProductUIModel;


import java.util.ArrayList;

public class ProductsAdapter extends RecyclerView.Adapter<ProductsAdapter.ProductViewHolder> {

    Context context;
    ArrayList<ProductUIModel> products;

    public ProductsAdapter(Context context, ArrayList<ProductUIModel> products) {
        this.context = context;
        this.products = products;
    }

    @NonNull
    @Override
    public ProductViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.product_item, parent, false);
        return new ProductViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ProductViewHolder holder, int position) {
        holder.productName.setText(products.get(position).getSku());
        holder.amount.setText(products.get(position).getAmount().toString());
    }

    @Override
    public int getItemCount() {
        return products.size();
    }

    public class ProductViewHolder extends RecyclerView.ViewHolder {

        TextView productName;
        TextView amount;

        public ProductViewHolder(@NonNull View itemView) {
            super(itemView);

            productName = itemView.findViewById(R.id.productName);
            amount = itemView.findViewById(R.id.productAmount);


        }
    }
}

File: presentation/src/main/java/com/example/presentation/products/adapter/ProductsAdapter.kt
package com.example.presentation.products.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.presentation.R
import com.example.presentation.model.ProductUIModel
import com.example.presentation.products.adapter.ProductsAdapter.ProductViewHolder
import kotlinx.android.synthetic.main.product_item.view.*
import java.util.*

class ProductsAdapter(
    var context: Context?,
    var products: ArrayList<ProductUIModel>,
    val clickListener: (ProductUIModel) -> Unit
) : RecyclerView.Adapter<ProductViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProductViewHolder {
        val view =
            LayoutInflater.from(context).inflate(R.layout.product_item, parent, false)
        return ProductViewHolder(view)
    }

    override fun onBindViewHolder(holder: ProductViewHolder, position: Int) {
        holder.bind(products[position], clickListener)
    }

    override fun getItemCount(): Int {
        return products.size
    }

    inner class ProductViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {


        fun bind(productUIModel: ProductUIModel, clickListener: (ProductUIModel) -> Unit) {
            itemView.productName.text = productUIModel.sku
            itemView.productAmount.text = productUIModel.amount.toString()
            itemView.setOnClickListener { clickListener(productUIModel) }
        }
    }

}